networkx/algorithms/d_separation.py

```diff
@@ -0,0 +1,122 @@
+"""
+Algorithm for testing d-separation in DAGs.
+
+*d-separation* is a test for conditional independence in probability
+distributions that can be factorized using DAGs.  It is a purely
+graphical test that uses the underlying graph and makes no reference
+to the actual distribution parameters.  For a detailed treatment see
+[1]_, [2]_.
+"""
+
+from collections import deque
+from typing import Hashable, AbstractSet
+
+import networkx as nx
+from networkx.utils import not_implemented_for, UnionFind
+
+__all__ = ["d_separated"]
+
+
+@not_implemented_for("undirected")
+def d_separated(G: nx.DiGraph, x: AbstractSet[Hashable],
+                y: AbstractSet[Hashable], z: AbstractSet[Hashable]) -> bool:
+    """
+    Test whether two sets of nodes are d-separated in a DAG by a third
+    set of conditioning nodes.
```

Comment:
To fulfill the 1-line summary requirement, here is a strawman suggestion.

```suggestion
    Return whether node sets ``x`` and ``y` are separated by ``z``.
```

Reply:
This is actually clear and concise.